NewForce-Cohort-7/tabloidfullstack-thegroundbeetles
# Awdeletecat

Description
Ticket https://github.com/NewForce-Cohort-7/tabloidfullstack-thegroundbeetles/pull/11 : Delete A Category & 12 Edit a Category

I added a feature to the page that allows you to delete and edit a category in the list.  If a user is logged in the app & clicks category then they will see a list of categories displayed on the page. Under each list item there will a delete and edit buttons will be displayed. A user can then click the delete button to delete a category. If they click the edit button they will be directed to a form where they can edit the selected category.



Type of change:
*New feature (non-breaking change which adds functionality)

Testing Instructions
Reference [How to test a teammate's PR](https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr)
git checkout AWdeletecat
If you don't see the most recent version of my files, git pull origin
AWdeletecat
Add all seed data by adding a new query and copying the code from the links below
Use this to fill out tables and columns:
https://github.com/NewForce-Cohort-7/tabloidfullstack-thegroundbeetles/blob/main/SQL/01_Tabloid_Create_DB.sql
Use this to seed the database:
https://github.com/NewForce-Cohort-7/tabloidfullstack-thegroundbeetles/blob/main/SQL/02_Tabloid_Seed_Data.sql
To execute your query just press Ctrl+Shift+e inside of your query.
Run the program by clicking the Green Arrow next to TabloidMVC at the top.
In the browser, please go to Login on the nav bar
Log in with this email for both the email and password:
[foo@bar.com](mailto:foo@bar.com)
This should direct you back to the home page.
Then you will click Category in the Navbar and list of categories should be displayed on the screen. Then click Edit Category (there should be a button next to each category) and you should be directed to a form that you can type a category name in to edit the category. Next click the save button and you should be directed back to the category list page. You should see your new category in the list and the list should be in alphabetical order. To Delete you just click the delete button next to the category you want to delete. The user will then be taken to a page that prompts, Are you sure you want to delete (category name you clicked delete button next to here)? And will be prompted to click a yes or no button. When no is clicked the user is directed back to the category list page and the category name remains on the list. If yes is clicked, the user is directed back to the category list page and the category is removed. 
Locate Logout on the Navbar & click this to log out.
Checklist:
[ x] - My code follows the style guidelines of this project
[x ] - I have performed a self-review of my code
[x ] - I have commented my code, particularly in hard-to-understand areas
[x ] - My changes generate no new warnings or errors
[x ] - I have added test instructions that prove that my feature works
